### PR TITLE
Add unit tests for cosine similarity package

### PR DESCRIPTION
**Pull request by LeoDev0**
## Summary

Adds the first tests for `backend/pkg/similarity/cosine.go` — the cosine similarity primitive used by `MemoryVectorStore.Search` and, transitively, by the entire RAG retrieval flow. Previously this package had no coverage, so a regression in the math would silently degrade retrieval quality without surfacing any failure.

### Changes

- **`backend/pkg/similarity/cosine_test.go`** — new file, table-driven tests following `backend/CLAUDE.md` conventions (testify, nested `input`/`expected` structs, no separator comments).
  - **`TestSearch`** covers: empty chunks list (early return), chunks with empty embeddings (skip path), descending-score ordering, `min(limit, len(scored))` cap in both directions, and the all-skipped edge case. Uses geometrically obvious vectors (identical / orthogonal / mid-similar) so expected ordering doesn't depend on float arithmetic.
  - **`TestCosineSimilarity`** pins the math directly with canonical cases (identical → 1.0, opposite → -1.0, orthogonal → 0.0) and the two defensive branches (length mismatch, zero-norm vectors). Uses `assert.InDelta` with `1e-9` epsilon to avoid float comparison flakiness. These math cases are not redundant with `TestSearch`: ordering tests could pass even with a broken formula by accident.

Result: **100% statement coverage** for the package.

## Test plan

- [x] `cd backend && go test -v -cover ./pkg/similarity/` reports `coverage: 100.0% of statements`
- [x] `cd backend && go test ./...` — full suite still green
- [x] `cd backend && go vet ./...`

### CODE REVIEW

**Review by LeoDev0 — COMMENTED** [review]
Solid coverage and structure — math is pinned directly, ordering tests use geometrically obvious vectors, and the package now reports 100% statement coverage. One follow-up below on `limit` boundary cases.

File: backend/pkg/similarity/cosine_test.go [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+				embedding: []float64{1, 0, 0},
+				chunks: []types.DocumentChunk{
+					{ID: "no-embedding-1"},
+					{ID: "no-embedding-2"},
+				},
+				limit: 5,
+			},
+			expected: expected{ids: []string{}},
+		},
+	}
```
Comment (LeoDev0) [inline backend/pkg/similarity/cosine_test.go:121]: Boundary cases for `limit` aren't covered. `limit == 0` is a clean addition (returns empty). `limit < 0` is more interesting — `min(-1, n) == -1`, so `scored[:k]` on `cosine.go:31` panics on any non-empty `scored`. Worth pinning that contract with `assert.Panics` here, or treating it as a latent bug and clamping `k` to a non-negative value in `cosine.go` (`if k < 0 { k = 0 }`).

Comment (LeoDev0) [reply backend/pkg/similarity/cosine_test.go:121]: Great review. Fixed.
